roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,161 @@
+            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+            if cmd.get("images", None):
+                found.add(image)
+
+        return found
+
+    def known_docker_registries(self, task_vars):
+        docker_info = self.module_executor("docker_info", {}, task_vars).get("info", "")
+        return [registry.get("Name", "") for registry in docker_info.get("Registries", {})]
+
+    def available_images(self, missing_images, registries, task_vars):
+        """ Inspect existing images using Skopeo and return all images successfully inspected """
+        missing_images_copy = list(missing_images)
+        registries_copy = list(registries)
+
+        skopeo_update = self.update_skopeo_image(task_vars)
+        changed = skopeo_update.get("changed", False)
+
+        # exit early if Skopeo update fails
+        if skopeo_update.get("failed", False):
+            return missing_images, changed
+
+        while registries_copy and missing_images_copy:
+            registry = registries_copy.pop(0)
+            # pylint: disable=redefined-variable-type
+            missing_images_copy = self.missing_skopeo_images(registry, missing_images_copy, task_vars)
+
+        return set(missing_images) - set(missing_images_copy), changed
+
+    def containerized_docker_images(self, base_name, version):
+        return [
+            "{image}:{version}".format(image=self.image_from_base_name(base_name), version=version)
+        ]
+
+    @staticmethod
+    def rpm_docker_images(base, version):
+        return [
+            "{image_repo}/registry-console:{version}".format(image_repo=base["repo"], version=version)
+        ]
+
+    @staticmethod
+    def qualified_docker_images(image_name, version):
+        # pylint: disable=unused-argument
+        # pylint: disable=unused-variable
+        return [
+            "{image_name}-{component}:{version}".format(**locals())
+            for component in "haproxy-router docker-registry deployer pod".split()
+        ]
+
+    @staticmethod
+    def image_from_base_name(base):
+        return "".join([base["repo"], "/", base["image"]])
+
+    def missing_skopeo_images(self, registry, tagged_images, task_vars):
+        """ Uses Skopeo to determine if required images exist in a given registry.
+         Returns the set of images not found in the given registry."""
+        failed = set()
+        for image in tagged_images:
+            cmd_str = "skopeo inspect docker://{registry}/{tagged_image}".format(
+                registry=registry,
+                tagged_image=image,
+            )
+            args = {
+                "name": "skopeo_inspect",
+                "image": self.skopeo_image,
+                "command": cmd_str,
+                "detach": False,
+            }
+
+            # the module's status will always be "1" if an image was not found
+            if self.module_executor("docker_container", args, task_vars).get("status") == 1:
```

Comment:
In all in-flight PRs, please review the places where we call `self.module_executor` -- we should be handling the case when calling the module fails (return value contains `{"failed": True}`).